### DISCUSSION TITLE
Disabling automatic screen analytics

### DISCUSSION THREAD

**Discussion by AlexanderIgnacz** [invertase/react-native-firebase#4355] [Q&A] [answered]
I've integrated successfully Firebase Analytics for screen tracks and I'd like to disable automatic screen analytics since I don't want to see the native screen classes including UIViewController from Firebase Analytics Events table.

- documentation regarding automatic analytics for screens.
https://firebase.google.com/docs/analytics/screenviews
- documentation regarding screen tracking with React Navigation.
https://reactnavigation.org/docs/screen-tracking/

Is there any way to disable automatic analytics of screen and enable only manual analytics?

**Comment by mikehardy** [comment, marked as answer]
I don't believe you can turn it off. You should use one of the recommended strategies based on your navigation system to set the screen correctly and log the javascript-relative screen names as the user navigates.

If you want no screen name/activity logged I believe you can follow the google docs by setting the screen name to null:

https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics#parameters_3

> The name of the current screen. Set to null to clear the current screen name.

**Reply by AlexanderIgnacz** [reply]
@mikehardy Thanks for your clarification. Could you give me example usage with `react-native-firebase` library? Currently I use another function since `setCurrentScreen()` is deprecated.
```
analytics()
            .logScreenView({ screen_name: currentRouteName, screen_class: currentRouteName })
```

**Reply by mikehardy** [reply]
No, sorry, you'll have to perform experiments locally and discover what works for you - it will end up being project-specific but our examples may be used as starters

**Reply by AlexanderIgnacz** [reply]
Ok, thanks. I found this useful description somehow.

```
Can I disable automatically collected screen_view events and only use manual logging?
Yes. While you can use both together at the same time, you can disable automatic event collection for screen_view events if you’d like. On iOS, set FirebaseAutomaticScreenReportingEnabled to “NO” in your info.plist. On Android, set google_analytics_automatic_screen_reporting_enabled to “false” in your manifest.
```

https://firebase.googleblog.com/2020/08/google-analytics-manual-screen-view.html
